lib/active_encode/engine_adapters/ffmpeg_adapter.rb

```diff
@@ -70,22 +73,13 @@ def find(id, opts = {})
         pid = get_pid(id)
         encode.input.id = pid if pid.present?
 
-        if File.file? working_path("error.log", id)
-          error = File.read working_path("error.log", id)
-          if error.present?
-            encode.state = :failed
-            encode.errors = [error]
-
-            return encode
-          end
-        end
-
-        encode.errors = []
-
         encode.current_operations = []
         encode.created_at, encode.updated_at = get_times encode.id
 
-        if running? pid
+        encode.errors = read_errors(id)
+        if encode.errors.present?
+          encode.state = :failed
```

Comment:
I think this makes sense but I wonder if we might ever be masking a time when the encode is still running but errors exist.  Maybe this isn't possible though with how the code is written?

Reply:
I thought about this, if error exists and the process is somehow running/stuck does the consumer care about that or just want to fail fast?